### PR TITLE
Adding Active Label Cleaning code

### PR DESCRIPTION
**Pull request by melanibe**
This PR adds `InnerEye-DataQuality` a stand-alone subfolder that contains associated to our pre-print ["Active label cleaning: Improving dataset quality under resource constraints"](https://arxiv.org/abs/2109.00574). 
In particular this subfolder provides the tools to:
1. Train noise robust models (co-teaching, ELR, SSL pretraining and finetuning capabilities)
2. Run the label cleaning simulation benchmark proposed in the above mentioned manuscript. 
3. Run the model selection benchmark.
4. All the code related to our benchmark datasets CIFAR10H and our proposed NoisyChestXray benchmark.

### CODE REVIEW

File: InnerEye-DataQuality/InnerEyeDataQuality/datasets/cifar10_utils.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        tf = tarfile.open(str(path_to_tar))
+        tf.extractall(local_path)
+        os.remove(path_to_tar)
+
+    return local_path_to_test_batch
+
+
+def get_cifar10_label_names(file: Optional[Path] = None) -> List[str]:
+    """
+    TBD
```
Comment (ant0nsc) [inline InnerEye-DataQuality/InnerEyeDataQuality/datasets/cifar10_utils.py:98]: can we either remove that completely, or fill in? Same for other occurrences of TBD

File: InnerEye-DataQuality/README.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
+# InnerEye-DataQuality
+
+TODO: Maybe name the root folder as `InnerEye-DataQuality` instead of DataSelection to make it consistent. And make the associated renaming in the files.
```
Comment (ant0nsc) [inline InnerEye-DataQuality/README.md:3]: pending TODO?

File: InnerEye-DataQuality/default_paths.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+
+from pathlib import Path
+
+PROJECT_ROOT_DIR = Path(__file__).parent.resolve()
+INNEREYE_DQ_DIR = PROJECT_ROOT_DIR / "InnerEyeDataQuality"
+CIFAR10_ROOT_DIR = Path("~/.torch/datasets/CIFAR10")
```
Comment (ant0nsc) [inline InnerEye-DataQuality/default_paths.py:10]: A folder in ~ looks suspicious, where is that coming from? Our choice?

File: azure-pipelines/build-pr.yml [hunk 4]
```diff
@@ -215,3 +215,10 @@ jobs:
           wait_for_completion: 'True'
           pytest_mark: ''
           max_run_duration: '30m'
+
+  # Now run the build for the Data Selection sub-folder
+  - job: BuildDataSelection
+    pool:
+      vmImage: 'ubuntu-18.04'
+    steps:
+      - template: build_data_quality.yaml
```
Comment (ant0nsc) [inline azure-pipelines/build-pr.yml:224]: Not sure we want to couple those two pipelines? We have a problem with long build time already, adding more bits on will make it even slower.

**Review by ant0nsc — APPROVED** [review]

File: azure-pipelines/build-pr.yml (hunk 4 above)
Comment (melanibe) [reply azure-pipelines/build-pr.yml:224]: It is actually a separate job that runs in parallel of all the other stages. It only takes 6 mins to run i.e. is one of the first to finish. Do you think this is a problem?

File: InnerEye-DataQuality/default_paths.py (hunk 3 above)
Comment (melanibe) [reply InnerEye-DataQuality/default_paths.py:10]: It is the default from torchvision. Updated to put it in a subfolder of the root now.

File: azure-pipelines/build-pr.yml (hunk 4 above)
Comment (melanibe) [reply azure-pipelines/build-pr.yml:224]: Resolved offline

**Review by ozan-oktay — APPROVED** [review]

**Review by ant0nsc — APPROVED** [review]
